src/bgw/job.c

```diff
@@ -597,9 +598,17 @@ bgw_job_tuple_update_by_id(TupleInfo *ti, void *const data)
 	BgwJob *updated_job = (BgwJob *) data;
 	HeapTuple tuple = heap_copytuple(ti->tuple);
 	FormData_bgw_job *fd = (FormData_bgw_job *) GETSTRUCT(tuple);
+	TimestampTz nowts, next_start;
 
 	ts_bgw_job_permission_check(updated_job);
-
+	/* when we update the schedule interval, modify the next start time as well*/
```

Comment:
I think this should only happen `if(fd->schedule_interval != updated_job->fd.schedule_interval)`

Reply:
changed.